### PR TITLE
Feature parsing 

### LINKED ISSUE #13: Enable forecasting parsing and types
Just for the record

### LINKED ISSUE #24: add thermal limit (or max rate) for all power lines.
When building a line or a transformer we need to make sure the rate is defined and below the SIL value. 

I am removing the ```Union{Nothing, Float64}``` in the rate field at branch types. 

see [cd32958](https://github.com/NREL/PowerSystems.jl/commit/cd329580a300d8686e28d63364585d4d086d0c22)

### LINKED ISSUE #29: Remove data and add build.jl
Create a deps file with all the data and remove data folder from the repository.

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Ability parse data from PowerModels to PowerSystems, fixes #13 
Removed data files and added build.jl to download data files from release based on tag,  fixes #29 
Add SIL limit check for lines , fixes #24 
Now only prints PowerModels errors while parsing, no more warnings

### CODE REVIEW

**Comment by jdlara-berkeley** [comment]
@sourabhdalvi, some of the testings, is failing, can you check where, please?

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,72 @@
+using InfoZIP
```
Comment (jdlara-berkeley) [inline deps/build.jl]: This line shuld be removed and use `InfoZip.X` instead.

File: src/models/topological_elements.jl [hunk 2]
```diff
@@ -19,7 +19,17 @@ Bus(;   number = 0,
         basevoltage = nothing
     ) = Bus(number, name, bustype, angle, voltage, PowerSystems.orderedlimits(voltagelimits, "Voltage"), basevoltage)
 
-struct LoadZones
+struct LoadZones  <: PowerSystemDevice
     number::Int
     name::String
+    buses::Array{PowerSystems.Bus,1}
+    maxrealpower::Float64
```
Comment (jdlara-berkeley) [inline src/models/topological_elements.jl]: Is this a maxrealpower transfer? @claytonpbarrows what is the common use of load zones?

File: src/models/forecasts.jl [hunk 3]
```diff
@@ -3,11 +3,12 @@ abstract type
 end
 
 struct Deterministic <: Forecast
+    device::PowerSystemDevice
```
Comment (jdlara-berkeley) [inline src/models/forecasts.jl]: @claytonpbarrows  should we add a field that specifies the forecast use? For instance Day Ahead vs Real Time?

File: src/models/topological_elements.jl (hunk 2 above)
Comment (claytonpbarrows) [reply src/models/topological_elements.jl]: no, I think it's peak load

File: src/models/forecasts.jl (hunk 3 above)
Comment (claytonpbarrows) [reply src/models/forecasts.jl]: I don't think so. I think that should be driven by some combination of the existing fields (which aren't completely clear to me). Basically we need to know:
 - What the forecast is for (device)
 - When the forecast is available (initial time?)
 - The horizon of the forecast (e.g. the timestamps for the values in the forecast timeseries). (interval, resolution, and horizon?)

**Pull request merged**